lib/cloudant.js

```diff
@@ -742,23 +736,32 @@ Cloudant.prototype.updateOrCreate = function(model, data, cb) {
   debug('Cloudant.prototype.updateOrCreate %j %j', model, data);
   var self = this;
   var idName = self.idName(model);
-  var id = data[idName];
+  var mo = self.selectModel(model);
+  var id = data[idName].toString();
+
+  // Callback handler for both create calls.
+  var createHandler = function(err, id) {
+    if (err) return cb(err);
+    mo.db.get(id, function(err, doc) {
+      if (err) return cb(err);
+      cb(err, self.fromDB(model, mo, doc), {isNewInstance: true});
+    });
+  };
+
   if (id) {
     self.updateAttributes(model, id, data, {}, function(err, docs) {
-      if (err && err.statusCode && err.statusCode === 404) {
-        self.create(model, data, {}, function(err) {
+      if (err && err.statusCode !== 404) return cb(err);
+      else if (err && err.statusCode && err.statusCode === 404) {
+        self.create(model, data, {}, createHandler);
+      } else {
+        mo.db.get(id, function(err, doc) {
```

Comment:
Does `updateAttributes` not return the resulting document after it's been changed? If it doesn't, then let's fix that instead.

Unless of course there's a Juggler-level API implication surrounding `updateAttributes`, where we don't give back the whole record again. :/